### ISSUE TITLE
Spurious "unreachable case branch" with multi-way conditional type

### ISSUE THREAD

**Issue by githorse** [microsoft/TypeScript#47651]
# Bug Report

### 🔎 Search Terms

unreachable case branch
switch statement
conditional type
multiple

### 🕗 Version & Regression Information

3.7.2

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about 3.7.2

### ⏯ Playground Link

[Playground link, kinda](https://www.typescriptlang.org/play?ts=3.6.3#code/C4TwDgpgBAKg7gewOoEMQB4YD4oF5ZQQAewEAdgCYDOUVwATgJZkDmUA-FAOR1OtcAubnACuwLgCgJAMxFkAxsEYIyUABYi1mLAAoiQ+MjTaAlFADeEgJBU4jYPLVQ9Zy1avyUVaDwbMWglD0EMAi9KoARACSXAC2UCi0fqwR1h5ePqLiQlC5waHhUNFxtAixIWr+hAA23qnuFBDSKCLVwDm5nflhkex99QC+EkMSoJCwasEQqBjYeATEpJQ0vFWcvnwBUEI6MIQk5NRQZCKxAEYQ9BzcJ+eXXNvCYlwmUgD0bxOXEDzHCIT0egIK7BMDA0gUdTfAA0UDOYigzCgUTIpGq1UYACkoEgIGcAMrAYGxCQfZFQFghBKROTBFCOFBnarQTzeJIoUjlVERKBwFDhfwAOmcUSgFBUXGAUFJn280HsiNUwDU0GAiAAtHyQFBYigwJBIWNoIyEAA3CCCsxRGjKjkyqDyMpVaRA+IwcAQKjyJhgYDsGRyRTKVRqWKxbR6AyTCDTYzYVzWWz2RzOIgJ9ysnyrfhCbqFYrxRLZlj1dJsri3C70QJ5yIxQvHU5V0uZp7ZTpBEI9Ir10rlZVVCC1CClxrNVrtDtdLv5vrsQbDKSOshUBDMwXVBAsHSh2I6Db+F5mKedMmXIH0CTL1frzfb3foYu6A-8ExmMkAeQA0kA)

The playground regrettably does not produce the error that I see in Webstorm, namely, an "unreachable 'case' branch" warning:

![image](https://user-images.githubusercontent.com/8271372/151583845-98151472-358b-4eb1-8c69-bb5f9d9a286b.png)

(Maybe that error's not coming from Typescript?)
Note that I do not see this switch statement error with a simpler two-way conditional type:

![image](https://user-images.githubusercontent.com/8271372/151584707-30b37357-ef5a-497a-bbf3-fb0e89788930.png)

### 💻 Code

```
type Foo<T> =
	  T extends string ? 'string'
	: T extends number ? 'number'
	: 'wut'

// all these work as expected
const w: Foo<string> = 'string' // ✓
const x: Foo<string> = 'number' // ✓
const y: Foo<number> = 'number' // ✓
const z: Foo<boolean> = 'wut'   // ✓

// but this shows a spurious error for an
// "unreachable 'case' branch"
function hmm<T>(x: Foo<T>) {
	switch (x) {
		case 'string': return "I'm a string"
		case 'number': return "I'm a number"
		case 'wut':    return "???"
		default:       return "dang it"
	}
}
```

### 🙁 Actual behavior

I see an "unreachable case branch" warning.

### 🙂 Expected behavior

No warnings.

**Comment by fatcerberus** [comment]
I'm pretty sure that's not a TypeScript error.

**Comment by DanielRosenwasser** [comment]
If it doesn't repro in the compiler, playground, VS Code, or VS, I have to assume it's external.
